portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -111,7 +117,7 @@ private int getNumberOfCommentsToDisplay(HttpServletRequest request) {
     int numComments;
 
     // Null or empty check.
-    if (requestValue == null || requestValue.isEmpty()) {
+    if ("null".equals(requestValue) || requestValue.isEmpty()) {
```

Comment:
I think it is still possible to get a real `null` value here. HttpServletRequest will return `null` if the requested parameter isn't there. So it would be safer to leave that test in as part of the set of disjunctions.
Even though you

Reply:
I think that you can get a `"null"` string value as the parameter because JavaScript will convert a `null` value to the string `"null"` if it gets coerced into a string. So when you build your URL query parameters with
```
const URL = '/data?commentLimit=' + commentLimit + '&languageCode=' + languageCode;
```
you can end up with the string `"null"` as the value, if the variable has a `null` value.

If you have time it would be better to handle this case by substituting the empty string or leaving the parameter out entirely. But at the same time, it would be a good defensive programming practice to handle both `null` and `"null"` values when you receive the parameters, since it would be possible for someplace other than your webpage to send traffic the web app if it were publicly exposed.

Reply:
Ok, will leave the check for an actual `null` value.